turbopack/crates/turbo-tasks-backend/src/backend/mod.rs

```diff
@@ -1997,8 +2004,9 @@ impl<B: BackingStorage> TurboTasksBackendInner<B> {
         ctx: &mut impl ExecuteContext<'_>,
         task_id: TaskId,
         new_children: FxHashSet<TaskId>,
-        is_now_immutable: bool,
     ) -> bool {
+        debug_assert!(!new_children.is_empty());
+
         let mut task = ctx.task(task_id, TaskDataCategory::All);
         let Some(in_progress) = get!(task, InProgress) else {
             panic!("Task execution completed, but task is not in progress: {task:#?}");
```

Comment:
is cancellation a concern here?  should this be a bail?

Reply:
We can't bail here, but yes I think we should handle cancellation by early returning instead of panic...